dotnet/ToDoApp/ToDoApp/ViewModel/WelcomeViewModel.cs

```diff
@@ -0,0 +1,184 @@
+using Realms;
+using Realms.Sync;
+using Xamarin.Forms;
+
+namespace ToDoApp
+{
+    // View model responsible for user sign up and sign in.
+    public class WelcomeViewModel : INotifyPropertyChanged
+    {
+        private string _message;
+        private string _username;
+        private string _password;
+        private bool _isFormEnabled = true;
+        private INavigation _navigation;
+
+        public string Username
+        {
+            get { return _username; }
+            set
+            {
+                if (_username == value)
+                {
+                    return;
+                }
+                _username = value;
+                OnPropertyChanged();
+            }
+        }
+
+        public string Password
+        {
+            get { return _password; }
+            set
+            {
+                if (_password == value)
+                {
+                    return;
+                }
+                _password = value;
+                OnPropertyChanged();
+            }
+        }
+
+        public string Message
+        {
+            get { return _message; }
+            set
+            {
+                if (_message == value)
+                {
+                    return;
+                }
+                _message = value;
+                OnPropertyChanged();
+            }
+        }
+
+        public bool IsFormEnabled
+        {
+            get { return _isFormEnabled; }
+            set
+            {
+                if (_isFormEnabled == value)
+                {
+                    return;
+                }
+                _isFormEnabled = value;
+                OnPropertyChanged();
+            }
+        }
+
+        public ICommand SignInCommand { get; private set; }
+        public ICommand SignUpCommand { get; private set; }
+
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        public WelcomeViewModel(INavigation navigation)
+        {
+            _navigation = navigation;
+            SignInCommand = new Command(() => SignIn(false).IgnoreResult());
+            SignUpCommand = new Command(() => SignIn(true).IgnoreResult());
+
+            // Check if a user is already logged in on this device.
+            var user = User.Current;
+            if (user != null)
+            {
+                // User is already logged in. Skip to the items page.
+                OpenRealmAndGoToItemsPage(user).IgnoreResult();
+            }
+        }
+
+        private async Task SignIn(bool createUser)
+        {
+            // Prevent input and display the status message.
+            IsFormEnabled = false;
+            Message = "Signing in...";
+
+            try
+            {
+                // Create credentials with the given username and password.
+                // Specify whether we are registering a new user or not with `createUser`.
+                var credentials = Credentials.UsernamePassword(Username, Password, createUser);
```

Comment:
If you don't pass anything for `createUser` (i.e. let it be `null`), the call will be interpreted as "login or register", where ROS will try to login a user if one exists with the specified username or register the user in a single API call. It can simplify the UI/example a little.